src/inmanta/env.py

```diff
@@ -618,17 +743,41 @@ def install_from_source(
     ) -> None:
         """
         Install one or more packages from source. Any path arguments should be local paths to a package directory or wheel.
+
+        This method provides backward compatibility with ISO6
         """
         if len(paths) == 0:
             raise Exception("install_from_source requires at least one package to install")
-        constraint_files = constraint_files if constraint_files is not None else []
-        inmanta_requirements = self._get_requirements_on_inmanta_package()
-        self._run_pip_install_command(
-            python_path=self.python_path,
+        self.install_for_config(
+            requirements=[],
             paths=paths,
-            constraints_files=constraint_files,
-            requirements=inmanta_requirements,
-            use_pip_config=True,
+            config=PipConfig(use_system_config=True),
+            constraint_files=constraint_files,
+        )
+
+    def install_from_list(
+        self,
+        requirements_list: Sequence[str],
+        *,
+        upgrade: bool = False,
+        upgrade_strategy: PipUpgradeStrategy = PipUpgradeStrategy.ONLY_IF_NEEDED,
+        use_pip_config: Optional[bool] = False,
+    ) -> None:
+        """
+        Install requirements from a list of requirement strings. This method uses the Python package repositories
+        configured on the host.
+        :param requirements_list: List of requirement strings to install.
+        :param upgrade: Upgrade requirements to the latest compatible version.
+        :param upgrade_strategy: The upgrade strategy to use for requirements' dependencies.
+        :param use_pip_config: Whether the pip config file specified in the PIP_CONFIG_FILE env var should be used
```

Comment:
This option was silently ignored: : https://github.com/inmanta/inmanta-core/blob/0e5eed689d1ba01fc02c2c070ccbd0a16171f248/src/inmanta/env.py#L506-L507

I.e. it was ignored if not index_url was set, which is not possible here.